src/Nancy/ModelBinding/DefaultBinder.cs

```diff
@@ -180,7 +180,7 @@ public object Bind(NancyContext context, Type modelType, object instance, Bindin
 
         private bool BindingValueIsValid(string bindingValue, object existingValue, BindingMemberInfo modelProperty, BindingContext bindingContext)
         {
-            return (!String.IsNullOrEmpty(bindingValue) &&
+            return ((!String.IsNullOrEmpty(bindingValue) || (!bindingContext.Configuration.ConvertEmptyStringToNull && bindingValue != null)) &&

```

Comment:
It's probably partly because it's late, but this boolean thingy right here is getting WAAAY too complex for my poor brain to parse. Think you could split it up into a couple of locals that has some good names? :smile: